modules/installation-aws-vpc-security-groups.adoc

```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws/installing-aws-government-region.adoc
+// * installing/installing_aws/installing-aws-secret-region.adoc
+// * installing/installing_aws/installing-aws-private.adoc
+// * installing/installing_aws/installing-aws-vpc.adoc
+
+:_content-type: PROCEDURE
+[id="installation-aws-vpc-security-groups_{context}"]
+= Applying existing security groups to the cluster
+By default, the installation program creates and attaches security groups to control plane and compute machines. The rules associated with the default security groups cannot be modified.
+
+If you are deploying an {product-title} cluster to an existing AWS Virtual Private Cloud (VPC), you can apply additional existing AWS security groups (custom security groups) to your control plane and compute machines. Applying custom security groups can help you meet the security needs of your organization, in such cases where you need to control the incoming and outgoing traffic of your control plane and compute machines.
+
+.Prerequisites
+* You have created the security groups in AWS. For more information, see the AWS documentation about working with link:docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security-groups.html[security groups].
+* The security groups must be part of the existing VPC to which you are deploying the cluster. The security groups cannot be associated with another VPC.
+* You have an existing `install-config.yaml` file.
+
+.Procedure
+
+. Edit the `compute.platform.aws.additionalSecurityGroupIDs` parameter to specify one or more custom security groups for your compute machines.
+. Edit `controlPlane.platform.aws.additionalSecurityGroupIDs` parameter to specify one or more custom security groups for your control plane machines.
+. Save the file and reference it when deploying the cluster.
+
+.Sample `install-config.yaml` files that specifies custom security groups
+[source,yaml]
+----
+#...
+compute:
+- hyperthreading: Enabled
+  name: worker
+  platform:
+    aws:
+      additionalSecurityGroupIDs:
+        - sg-1
+        - sg-2
+  replicas: 3
```

Comment:
I think it's important to make sure that feature also can work with the edge compute pool with Local Zones.
https://docs.openshift.com/container-platform/4.13/installing/installing_aws/installing-aws-localzone.html

@barbacbd can you see any restriction on that?

Reply:
@mtulio as far as local zones go, security groups don't require anything special. The security groups just need to be attached to the VPC with the local zone subnet. Also we would require that these local zone subnets are provided in the subnets field of the install config.

Reply:
@mtulio as far as edge computing, what concerns do you have? Security groups can be shared by many resources as they are just rules.

Reply:
> @mtulio as far as edge computing, what concerns do you have? Security groups can be shared by many resources as they are just rules.

Yes, SGs can be assigned to any supported resource between the same VPC.

My concern was on the install side, I just wanna make sure the configuration below can be used - considering `edge` compute pool uses the same flow of `worker`, I expect only `sg-5` and `sg-6` in MachineSet manifests for Local Zones, just confirming it.

```yaml
compute:
- hyperthreading: Enabled
  name: worker
  platform:
    aws:
      additionalSecurityGroupIDs:
        - sg-1 <1>
        - sg-2
  replicas: 3
- hyperthreading: Enabled
  name: edge
  platform:
    aws:
      additionalSecurityGroupIDs:
        - sg-5
        - sg-6
```

Reply:
Confirmed! The new machineset does show that it is for edge nodes and it has the security group that I mentioned in the install config and not the one used for the workers